atomics/include/desul/atomics/Compare_Exchange_CUDA.hpp

```diff
@@ -18,15 +18,15 @@ namespace desul {
 template <typename T, class MemoryScope>
 __device__ typename std::enable_if<sizeof(T) == 4, T>::type atomic_compare_exchange(
     T* const dest, T compare, T value, MemoryOrderRelaxed, MemoryScope) {
-  unsigned int return_val = atomicCAS(reinterpret_cast<unsigned int*>(dest),
+  uint32_t return_val = atomicCAS(reinterpret_cast<unsigned int*>(dest),
```

Comment:
I should have done the rest. Using int, or unsigned int, as any specific size is unwise. Why would you be emphatic about leaving it ill-specified?

Reply:
Actually, I think I'll do that, these should all use fixed sizes, and if they get something that can't support that size they should fail to compile IMO, it would be really bad to get a mismatch there.

Reply:
I ended up doing this one differently, more the way you and Christian had it.  The typedefs for uint64_t (or whatever they are) didn't act type compatible, so I put all of them back to using vaguely sized types, but with appropriate static_asserts to ensure we don't miss an issue if one crops up at some point.